lua/weapons/weapon_snowball.lua

```diff
@@ -0,0 +1,133 @@
+SWEP.PrintName = "Snowballs"	
+
+SWEP.Instructions = "Left click to throw a snowball\nRight click changes trail color"
+
+SWEP.ViewModel = "models/weapons/v_snowball.mdl"
+SWEP.WorldModel = "models/weapons/w_snowball.mdl" 
+
+SWEP.Slot = 4
+SWEP.SlotPos = 1
+
+SWEP.DrawAmmo = false
+SWEP.DrawCrosshair = false
+SWEP.ViewModelFOV = 68
+
+SWEP.AutoSwitchTo = false
+SWEP.HoldType = "grenade"
+
+SWEP.Category = "Snowballs"
+SWEP.Spawnable = true
+
+SWEP.Primary.ClipSize = 1
+
+SWEP.Primary.Automatic = true
+SWEP.ThrowSound = Sound("weapons/iceaxe/iceaxe_swing1.wav")
+SWEP.ReloadSound = Sound("weapons/weapon_snowball/crunch.ogg")
+
+local ti = os.date("%B", os.time())
+if ti == "December" then --only activate during December
+	hook.Add("PlayerChangeLocation", "ChristmasSnowballs", function(ply, loc, old) --auto equip the snowball when outside
+		if !IsValid(ply) then return end
+		
+		local locname = GetLocationNameByIndex(loc)
+
+		if locname == "Outside" or locname == "Golf" and !ply:HasWeapon("weapon_snowball") then
+			ply:Give("weapon_snowball")
+		end
+	end)
+end
+
+if SERVER then --network the player's new color
+	util.AddNetworkString("CLtoSVSnowballColor")
+	util.AddNetworkString("SVtoCLSnowballColor")
+	net.Receive("CLtoSVSnowballColor", function(len, ply)
+		local col = net.ReadTable()
+		ply.currentcolor = col
+		net.Start("SVtoCLSnowballColor")
+			net.WriteTable(col)
+			net.WriteEntity(ply)
+			net.Broadcast()
+	end)
+else
+	net.Receive("SVtoCLSnowballColor", function()
+		local col = net.ReadTable()
+		local ply = net.ReadEntity()
+		if !IsValid(ply) then return end
+		ply.currentcolor = col
+	end)
+end
+
+function SWEP:Initialize()
+	self:SetWeaponHoldType(self.HoldType)
+	self.Weapon:SetClip1(1)
+	self.Owner.currentcolor = self.Owner.currentcolor or Color(255, 255, 255)
```

Comment:
"currentcolor" is set on the enttable of the player itself, so it should be something like "snowballcolor" to prevent conflicts (but in this case please use NWVector instead, but put "snowball" in the name of the vector for the same reason)

Reply:
A NWVector will cause the alpha channel to not do anything, but that should be fine

Reply:
Wait why is there an alpha channel? lol

Reply:
i was thinking of having an alpha channel if players didn't want trails, but the same effect can be achieved by just setting the trail color to black, so it's fine